scripts/gen_self_signed_cert.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+from cryptography import x509
+from cryptography.x509.oid import NameOID
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography.hazmat.primitives.asymmetric import rsa
+
+from datetime import datetime, timedelta
+
+cert_file_name = "./certificate.pem"
+key_file_name = "./key.pem"
+passphrase = b"passphrase"
+
+# Generate our key
+key = rsa.generate_private_key(
+    public_exponent=65537,
+    key_size=2048,
+    backend=default_backend()
+)
+# Write our key to disk for safe keeping
+with open("./key.pem", "wb") as f:
+    f.write(key.private_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=serialization.BestAvailableEncryption(passphrase),
+    ))
+# Various details about who we are. For a self-signed certificate the
+# subject and issuer are always the same.
+subject = issuer = x509.Name([
+    x509.NameAttribute(NameOID.COUNTRY_NAME, u"US"),
+    x509.NameAttribute(NameOID.STATE_OR_PROVINCE_NAME, u"WA"),
+    x509.NameAttribute(NameOID.LOCALITY_NAME, u"Redmond"),
+    x509.NameAttribute(NameOID.ORGANIZATION_NAME, u"ContosoDoNotUse"),
+    x509.NameAttribute(NameOID.COMMON_NAME, u"self-signed-do-not-use"),
+])
+cert = x509.CertificateBuilder().subject_name(
+    subject
+).issuer_name(
+    issuer
+).public_key(
+    key.public_key()
+).serial_number(
+    x509.random_serial_number()
+).not_valid_before(
+    datetime.utcnow()
```

Comment:
Just out of curiosity, have you verified that this works?  If so, I am very happy.
As a broader question, does the CertificateBuilder object have a dependency on the openssl executable like Node does?  Please say that it doesn't.

Reply:
I have verified this. I could send messages :)
WRT to openssl: the package does have references to openssl headers and says this in the FAQ:
Installing cryptography with OpenSSL 0.9.8 or 1.0.0 fails
The OpenSSL project has dropped support for the 0.9.8 and 1.0.0 release series. Since they are no longer receiving security patches from upstream, cryptography is also dropping support for them. To fix this issue you should upgrade to a newer version of OpenSSL (1.0.2 or later). This may require you to upgrade to a newer operating system.

So it may need at least the library - not sure about the executable themselves...

---
In reply to: [238395103](https://github.com/Azure/azure-iot-sdk-python-preview/pull/29#discussion_r238395103) [](ancestors = 238395103)